File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/exchange-rate-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { Currency } from "../enums/currency";

export interface ExchangeRateDto extends FullAuditedEntityDto<number> {
    currency?: Currency;
    baseCurrency?: Currency;
    baseToRate: number;
    dateEffected: Date;
}

File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/ExchangeRateDto.cs
using ProjectPos.Data.Shared.Enums;
using ProjectPos.Services.EntityDtos;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectPos.Services.DTOs
{
    public class ExchangeRateDto : FullAuditedEntityDto<int>
    {
        public Currency? Currency { get; set; }
        public Currency? BaseCurrency { get; set; }
        public decimal BaseToRate { get; set; }
        public DateTime DateEffected { get; set; }
    }
}
